pkg/server/server.go

```diff
@@ -20,35 +25,64 @@ type Server struct {
 func handleRequest(conn net.Conn) {
 }
 
-func handleClient(conn net.Conn) {
-	handshake := proto.Handshake{Name: "wavesT",
-		VersionMajor:      0x0,
-		VersionMinor:      0xe,
-		VersionPatch:      0x4,
-		NodeName:          "gowaves",
-		NodeNonce:         0x0,
-		DeclaredAddrBytes: []byte{},
-		Timestamp:         uint64(time.Now().Unix())}
-
-	_, err := handshake.WriteTo(conn)
-	if err != nil {
-		zap.S().Error("failed to send handshake: ", err)
-		return
-	}
-	_, err = handshake.ReadFrom(conn)
-	if err != nil {
-		zap.S().Error("failed to read handshake: ", err)
-		return
+func dialContext(ctx context.Context, network, addr string) (net.Conn, error) {
+	var major, minor, patch uint32
+	minor = 5
```

Comment:
this is a constant. don`t you think so?

Reply:
no, the idea behind this is following: there is a default ```DialContext``` implementation in the transport that just wraps the ```net.Dialer.DialContext``` and tries to connect to a node.
Here we redefine the ```DefaultTransport.DialContext``` behaviour to try to re-adjust the connection number of the node. So, for instance, if the version of the node we are trying to connect to is ```0.5.5``` we begin with trying to connect with version ```0.5.10```, the connection gets terminated, we try ```0.5.9``` and so on. Most likely we will have to connect to a large variety of combinations of Major.Minor.Patch versions. Up until this day only the Major version is constant and for every existing node in the world is ```0```.